File: API/Model/UserModel.cs
namespace API.Model
{
    public class UserModel
    {
        //User Info
        public long UserId { get; set; }
        public string UserName { get; set; } = string.Empty;
        public string PasswordHash { get; set; } = string.Empty;
        public string Role { get; set; } = string.Empty; // Admin, Student, Company
        public string Status { get; set; } = "Active"; // Active, Pending, Rejected
        public string? ProfileImagePath { get; set; }

        //Contact Info
        public string Email { get; set; } = string.Empty;
        public string MobileNo { get; set; } = string.Empty;
        public string StreetAddress { get; set; } = string.Empty;
        public string City { get; set; } = string.Empty;
        public string District { get; set; } = string.Empty;
        public string State { get; set; } = string.Empty;
        public string Country { get; set; } = string.Empty;
        public long PinCode { get; set; }
        public bool IsDeleted { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation
        public StudentModel? Student { get; set; }
        public CompanyModel? Company { get; set; }
    }
    public class UserCreateDTO
    {
        public IFormFile? Resume { get; set; }
        public IFormFile? ProfilePhoto { get; set; }
        public string? Data { get; set; }
    }

    public class ChangePasswordRequest
    {
        public long UserId { get; set; }
        public string OldPassword { get; set; } = string.Empty;
        public string NewPassword { get; set; } = string.Empty;
        public string VerifyNewPassword { get; set; } = string.Empty;
    }
}

File: UI/src/app/shared/models/user.model.ts
import { CompanyModel } from "./company.model";
import { StudentModel } from "./student.model";

export class UserModel {
    isLoggedIn: boolean = false;

    // User Info
    userId: number = 0;
    userName: string = '';
    passwordHash: string = '';
    role: string = ''; // Admin, Student, Company
    status: string = 'Active'; // Active, Pending, Rejected
    profileImagePath?: string;
    profilePhoto?: File | null = null;

    // Contact Info
    email: string = '';
    mobileNo: string = '';
    streetAddress: string = '';
    city: string = '';
    district: string = '';
    state: string = '';
    country: string = '';
    pinCode: number = 0;

    isDeleted: boolean = false;
    createdAt: Date = new Date();
    updatedAt: Date = new Date();

    // Navigation
    student?: StudentModel | null = new StudentModel();
    company?: CompanyModel | null = new CompanyModel();

}
